### ISSUE TITLE
getter-return does not consider flow analysis

### ISSUE THREAD

**Issue by luvies** [denoland/deno_lint#168]
Using the following example:

```ts
export class Test {
  #internal: string | undefined;

  constructor(testStr?: string) {
    this.#internal = testStr;
  }

  get str(): string {
    if (this.#internal !== undefined) {
      return this.#internal;
    } else {
      return "empty";
    }
  }
}
```

I get the following output:

```
(getter-return) Getter requires a return
  get str(): string {
                    ^
    at test.ts:8:20

Found 1 problems
```

As a side-note, this rule is only useful in JS, as the TypeScript compiler errors with the following if a getter does not return:

`A 'get' accessor must return a value. ts(2378)`

Is it worth keeping the rule in recommended?

**Comment by disizali** [comment]
Thank you for reporting this bug,
I will push the fix in a few minutes.

> as the TypeScript compiler

**javascript**, and typescript actually, so this rule must remain where it is IMO.
